lib/manageiq/messaging/stomp/common.rb

```diff
@@ -18,6 +18,7 @@ def queue_for_publish(options)
           headers[:AMQ_SCHEDULED_TIME] = options[:deliver_on].to_i * 1000 if options[:deliver_on]
           headers[:priority]           = options[:priority] if options[:priority]
           headers[:_AMQ_GROUP_ID]      = options[:group_name] if options[:group_name]
+          headers[:encoding]           = options[:encoding] if options[:encoding]
```

Comment:
Maybe ok to accept encoding from options, but it is not in effect. `#encode_body` does not depend on it, and user's consumer code does not get the header either.

Reply:
I added tests to test this case.

This is when you encode the message yourself and want to send a custom encoding value across.